experiment/simulation/js/components.js

```diff
@@ -0,0 +1,113 @@
+
+const jsplumbInstance = jsPlumb.getInstance({
+
+    container: diagram,
+    maxConnections: -1,
+    endpoint: {
+        type: "Dot",
+        options: { radius: 7 },
+    },
+    dragOptions: {
+        containment: "parentEnclosed",
+        containmentPadding: 5,
+    },
+    connector: "Flowchart",
+    paintStyle: { strokeWidth: 3, stroke: "#456" },
+    connectionsDetachable: true,
+});
+jsplumbInstance.bind("ready", function () {
+    jsplumbInstance.registerConnectionTypes({
+        "red-connection": {
+            paintStyle: { stroke: "red", strokeWidth: 3 },
+            hoverPaintStyle: { stroke: "red", strokeWidth: 8 },
+            connector: "Flowchart"
+        }
+    });
+});
+
+
+jsplumbInstance.bind("connection", () => {
+    const con = jsplumbInstance.getAllConnections();
+    connectionMap.clear();
+    for (i = 0; i < con.length; i++) {
+        const s = con[i].sourceId, t = con[i].targetId;
+        const r = s.concat("$", t);
+        connectionMap.set(r, t)
+
+    }
+});
+jsplumbInstance.bind("dblclick", function (ci) {
+
+    jsplumbInstance.deleteConnection(ci);
+    const con = jsplumbInstance.getAllConnections();

```

Comment:
```
const con = jsplumbInstance.getAllConnections();
     connectionMap.clear();
     for (i = 0; i < con.length; i++) {
         const s = con[i].sourceId, t = con[i].targetId;
         const r = s.concat("$", t);
         connectionMap.set(r, t)

     }
     console.log(connectionMap)```

you use the same piece in two places, just make this into a separate function and call that function twice.

Reply:
Now made a common function